updatehub-sdk/src/listener.rs

```diff
@@ -73,6 +77,17 @@ impl StateChangeListener {
         StateChangeListener::default()
     }
 
+    /// Function that register callback(s) for the state(s).
+    /// # Example
+    ///
+    /// ```text
```

Comment:
This should be a `no_run` block with the irrelevant parts hidden so the compiler can still check them.